### PR TITLE
fix(DatePickerInput): disabled state of years in case if they is limit

### PR DESCRIPTION
**Pull request by AregSoft**

### CODE REVIEW

**Review by NairaMisakyan — APPROVED** [review]

File: src/lib/organisms/DatePicker/Years/index.js [hunk 1]
```diff
@@ -2,19 +2,42 @@ import React, { useMemo, useCallback } from 'react';
...
+                return year >= dayjs(min).year();
+            }
+            if (!min && max) {
+                return year <= dayjs(max).year();
+            }
+            if (min && max) {
+                return max && year <= dayjs(max).year() && min && year >= dayjs(min).year();
+            }
+
+            return true;
```
Comment (GaroGabrielyan) [inline src/lib/organisms/DatePicker/Years/index.js:34]: You can simplify here 
           ```
 if (!year) return false;
            if (min && !max)  return year >= dayjs(min).year();
            if (!min && max)  return year <= dayjs(max).year();
            if (min && max) return max && year <= dayjs(max).year() && min && year >= dayjs(min).year();
            return true;
```

**Review by NairaMisakyan — APPROVED** [review]

**Review by GaroGabrielyan — DISMISSED** [review]

File: src/lib/organisms/DatePicker/Years/index.js [hunk 2]
```diff
@@ -6,15 +6,28 @@ import dayjs from 'dayjs';
...
+// Components
+import Button from '../../../atoms/Button';
+
+function Years({ previewYear, selected, onChange, onPreviewChange, max, min }) {
     const startOfDecade = useMemo(() => getStartOfDecade(previewYear), [previewYear]);
     const years = useMemo(() => chunk(getCalendarYears(startOfDecade), 3), [startOfDecade]);
 
     const isFromDecade = useCallback((year) => year > startOfDecade - 1 && year < startOfDecade + 10, [startOfDecade]);
 
+    const isYearMaxOrMinChecker = useCallback(
```
Comment (hamikhambardzumyan) [inline src/lib/organisms/DatePicker/Years/index.js:18]: Rename the function e.g. `checkYearDisableState`

File: src/lib/organisms/DatePicker/Calendar/index.js [hunk 3]
```diff
@@ -91,6 +91,16 @@ const Calendar = forwardRef((props, ref) => {
         }
     }, [value]);
 
+    useEffect(() => {
+        if (min && !max) setIsThisDateAllowed(dayjs(min).isSameOrBefore(dayjs(new Date()), 'date'));
+        if (!min && max) setIsThisDateAllowed(dayjs(max).isSameOrAfter(dayjs(new Date()), 'date'));
+        if (min && max)
+            setIsThisDateAllowed(
+                dayjs(min).isSameOrBefore(dayjs(new Date()), 'date') &&
```
Comment (hamikhambardzumyan) [inline src/lib/organisms/DatePicker/Calendar/index.js:99]:
```suggestion
                dayjs(min).isSameOrBefore(dayjs(), 'date') &&
```

File: src/lib/organisms/DatePicker/Calendar/index.js [hunk 4]
```diff
@@ -91,6 +91,16 @@ const Calendar = forwardRef((props, ref) => {
...
+                    dayjs(max).isSameOrAfter(dayjs(new Date()), 'date')
```
Comment (hamikhambardzumyan) [inline src/lib/organisms/DatePicker/Calendar/index.js:100]:
```suggestion
                    dayjs(max).isSameOrAfter(dayjs(), 'date')
```

File: src/lib/organisms/DatePicker/Calendar/index.js [hunk 5]
```diff
@@ -91,6 +91,16 @@ const Calendar = forwardRef((props, ref) => {
         }
     }, [value]);
 
+    useEffect(() => {
+        if (min && !max) setIsThisDateAllowed(dayjs(min).isSameOrBefore(dayjs(new Date()), 'date'));
```
Comment (hamikhambardzumyan) [inline src/lib/organisms/DatePicker/Calendar/index.js:95]:
```suggestion
        if (min && !max) setIsThisDateAllowed(dayjs(min).isSameOrBefore(dayjs(), 'date'));
```

File: src/lib/organisms/DatePicker/Calendar/index.js [hunk 6]
```diff
@@ -91,6 +91,16 @@ const Calendar = forwardRef((props, ref) => {
...
+        if (!min && max) setIsThisDateAllowed(dayjs(max).isSameOrAfter(dayjs(new Date()), 'date'));
```
Comment (hamikhambardzumyan) [inline src/lib/organisms/DatePicker/Calendar/index.js:96]:
```suggestion
        if (!min && max) setIsThisDateAllowed(dayjs(max).isSameOrAfter(dayjs(), 'date'));
```

**Review by hamikhambardzumyan — CHANGES_REQUESTED** [review]

**Review by hamikhambardzumyan — APPROVED** [review]

**Review by NarekArshakyan — APPROVED** [review]

**Pull request merged**